simulator/mips/mips_instr.cpp

```diff
@@ -483,5 +483,14 @@ void BaseMIPSInstr<RegisterUInt>::check_trap()
     }
 }
 
+template<typename RegisterUInt>
+void BaseMIPSInstr<RegisterUInt>::set_sequence_id( uint64 id )
+{
+    sequence_id = id;
+    std::ostringstream oss;
+    oss << "\t{" << id << "}";
+
```

Comment:
It does not append "disasm"...

Reply:
ok, i will look for the examples in mips_instr.cpp better

Reply:
should it be 
```
std::ostringstream oss;
491     oss << "\t{" << id << "}";
492     disasm += oss.str();
```
?

Reply:
Build and run the simulator and see whether it looks good for you.

Reply:
It's ok, but I don't understand why only one instruction is dumped
```
yan@yan-X555LJ:~/mipt-mips/simulator$ ./mipt-mips -b ../traces/factorial.out -n 100
Checker mismatch: Checker output: 0x4000f0: lui $t0, 0x41	{0}	 [ $t0 = 0x410000 ]
PerfSim output: 0x4000f0: lui $t0, 0x41	 [ $t0 = 0x410000 ]
```

Reply:
That's an output of "Checker mismatch" error.

We have a functional simulator (checker) running aside of a performance simulator, and it checks whether the same instructions perform writeback in a same order. The comparison is performed with `FuncInstr::is_same` function we discussed previously.

If you have already added sequence_id comparison to that function, that is the case you have to fix: functional simulator has "0", and performance simulator has "NO_VAL64".

If you want to run functional simulator only, use `-f` flag as it is stated in README.
If you want simulator to dump all the instructions, use `-d` option.

Reply:
Can I initialize sequence_id with "0" in performance sim?

Reply:
in BaseMIPSInstr and RISCVInstr class

Reply:
> If you have already added sequence_id comparison to that function
Please clarify first, do you have the change in the code you compiled and run when you got the error report shared above?

> Can I initialize sequence_id with "0" in performance sim?
For instruction # 0, yes. For instruction # 1 — no.

Reply:
I don't have added sequence_id comparison